File: University_Documentations/WeatherApp/app/src/main/java/com/example/weatherapp/RecyclerAdapter.java
package com.example.weatherapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.lang.ref.WeakReference;
import java.text.DateFormatSymbols;
import java.util.ArrayList;
import java.util.Calendar;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ViewHolder> {
    WeatherInformation[] weatherInformations;
    LayoutInflater inflater;

    WeakReference<Context> mainContext;

    public RecyclerAdapter(Context context, WeatherInformation[] weatherInformations) {
        inflater = LayoutInflater.from(context);
        mainContext = new WeakReference<>(context);
        this.weatherInformations = weatherInformations;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }


    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.otherday_weather_condition_recycler_row, parent, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        WeatherInformation weatherInformation = weatherInformations[position];

        int drawableResourceId = holder.itemView.getResources().getIdentifier("weathericon_".concat(weatherInformation.getIconName()), "drawable", mainContext.get().getPackageName());
        ImageView weatherIcon = holder.itemView.findViewById(R.id.imageViewOtherDayWeatherConditionIcon);
        weatherIcon.setImageResource(drawableResourceId);

        TextView maxTempTextView = holder.itemView.findViewById(R.id.textViewOtherDayWeatherConditionMaxTemp);
        TextView minTempTextView = holder.itemView.findViewById(R.id.textViewOtherDayWeatherConditionMinTemp);
        TextView weatherConditionNameTextView = holder.itemView.findViewById(R.id.textViewotherDayWeatherConditionName);
        TextView dayNameTextView = holder.itemView.findViewById(R.id.textViewDayName);

        String dayNames[] = new DateFormatSymbols().getWeekdays();
        int maxTemp = (int) weatherInformation.getMaxTemp();
        int minTemp = (int) weatherInformation.getMinTemp();

        maxTempTextView.setText(Integer.toString(maxTemp).concat("°"));
        minTempTextView.setText(Integer.toString(minTemp).concat("°"));
        weatherConditionNameTextView.setText(weatherInformation.getName());
        dayNameTextView.setText(dayNames[weatherInformation.getCalendar().get(Calendar.DAY_OF_WEEK)]);

    }

    @Override
    public int getItemCount() {
        return weatherInformations.length;
    }

}

File: SuperHeroBookRecyclerView/app/src/main/java/com/example/superherobookrecyclerview/RecyclerAdapter.kt
package com.example.superherobookrecyclerview


import android.content.Intent
import android.graphics.Bitmap
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder

class RecyclerAdapter(val superHeroList:ArrayList<String>,val superHeroImages:ArrayList<Int>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    class ViewHolder(view: View) : RecyclerView.ViewHolder(view){

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        // Inflater , LayoutInflater, MenuInflater

        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.recycler_row,parent,false)
        return ViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return superHeroList.size
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {

        val textView = holder.itemView.findViewById<TextView>(R.id.recyclerViewTextView)
        textView.setText(superHeroList[position])
        textView.setOnClickListener{
            val intent = Intent(holder.itemView.context,IntroductionActivity::class.java)
            intent.putExtra("superHeroName",superHeroList[position])
            intent.putExtra("superHeroImage",superHeroImages[position])
            holder.itemView.context.startActivity(intent)
        }
    }

}